src/database/UIDatabase.js

```diff
@@ -262,9 +262,14 @@ class UIDatabase {
     return this.database.write(...args);
   }
 
+  getPreference(key) {
+    const preference = this.database.get('Pref', key);
+    return preference?.value;
+  }
+
   getSetting(key) {
     const setting = this.database.get('Setting', key, 'key');
-    return (setting && setting.value) || '';
+    return setting?.value ?? '';
```

Comment:
not part of the issue, please revert

Reply:
gonna say the opposite - nice!

Reply:
I don't see the problem - it's a much neater implementation

Reply:
So does that mean, anytime we are working on an issue and editing a file that can be nice and neater with use of optional chaining and null coalescing operator, we should put an effort into finding and converting to that syntax ?

I really want to link some PRs here (desktop sync) that all four of us have reviewed where all four of us concluded that 'surgical' and 'targeted' approach to code change should have been exercised (it's been a soft PR 'guideline' in desktop for awhile).

(not saying I don't like the syntax btw, even if technically `??` is not the same as `||`)

Reply:
> So does that mean, anytime we are working on an issue and editing a file that can be nice and neater with use of optional chaining and null coalescing operator, we should put an effort into finding and converting to that syntax ?

ya!

> I really want to link some PRs here (desktop sync) that all four of us have reviewed where all four of us concluded that 'surgical' and 'targeted' approach to code change should have been exercised (it's been a soft PR 'guideline' in desktop for awhile).

I agree with it on desktop - should touch as little as possible haha that's the wild west. But doing refactors is how you avoid that situation and since we don't really do scheduled refactors, I think as you go is a great solution. I think even bigger refactors in a PR are all good if ya mention it in the PR comment!

Reply:
yep. agree that in desktop we should be super careful and not go wild with refactoring. We're planning to deprecate that codebase and there isn't a test framework to give us any safety when refactoring.

Also agree that in small and specific cases that it is actually good to do in an active codebase. Keeps it clean. Yes, the danger is that you introduce bugs and those sort of bugs are actually the hardest to find later because they don't directly relate to the issue.

In this case though- the change is simple, functionally equivalent and in an area that will be directly tested by the testing for this issue.

Reply:
I am glad we having this conversation actually, because I was and still am of a different opinion.  

@joshxg 
> I agree with it on desktop - should touch as little as possible haha that's the wild west. But doing refactors is how you avoid that situation

I haven't seen any evidence that refactoring of code outside of issues scope is how we avoid that situation. I have seen cases where refactoring of code outside of issue scope cased regression

>  I think even bigger refactors in a PR are all good if ya mention it in the PR comment!

Why would it be all good if it's not solving the issue it addresses ?

@mark-prins 
> We're planning to deprecate that codebase and there isn't a test framework to give us any safety when refactoring.

I don't believe we have one in mobile too, and are starting to get quite a few permutation of preferences / settings, which is already making testing harder.

> Also agree that in small and specific cases that it is actually good to do in an active codebase. Keeps it clean.

If there is code that is obviously needs a clean up, we can make an issue to clean it up. I don't actually agree that existing code was not clean (yes it neater and nicer with the change though, but that's not saying it wasn't clean, or not safe or less readable before)

> In this case though- the change is simple, functionally equivalent and in an area that will be directly tested by the testing for this issue.

The change itself is simple yes, and it's functionally equivalent  (even though `??'` is not quite the same as `||`) testing wise however, it's not part of preferences testing and this issue testing. 

Btw, I am pretty sure optional chaining was made for deeper then one nested object property check, and there was no need to use null coalescent here, '||' would look just as neat and further preserves functionality.

My conclusion is still
* Don't touch things that are not solving the issue, the focus should be on solving the issue, not making code base neater and nicer.
* If you notice something 'wrong' with existing code, make another issue for it
* In existing code where things are obviously not broken (and look perfectly fine), It should be up to stakeholders to decide if it's worth refactoring (issue)

Sounds like this conversation is getting too big for this issue, we should archive it somewhere (like wiki) once we reached a consensus.

By the way, this PR has refactoring (great refactoring) which is within the scope of the issue, which are completely fine.

And lastly, I do realise I am spending quite a bit of time commenting on one line of code, but it's beyond that actually, it's one area of PR 'guidelines' that needs to be agree on and clarified (by us 4)

Reply:
I'll move this to another issue, resolving conversation and 2vs1 so will approve this change